packages/browser/__tests__/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -48,33 +48,20 @@ describe("ClientRegistrar", () => {
   }
 
   describe("getClient", () => {
-    it("returns a formatted client object if the clientId is present", async () => {
-      const clientRegistrar = getClientRegistrar();
-      expect(
-        await clientRegistrar.getClient(
-          {
-            sessionId: "mySession",
-            redirectUrl: new URL("https://example.com"),
-            clientId: "coolApp",
-          },
-          IssuerConfigFetcherFetchConfigResponse
-        )
-      ).toMatchObject({
-        clientId: "coolApp",
-      });
-    });
-
     it("properly performs dynamic registration", async () => {
-      defaultMocks.fetcher.fetch.mockResolvedValueOnce(
+      const mockFetch = jest.fn().mockResolvedValueOnce(
         /* eslint-disable @typescript-eslint/camelcase */
         (new NodeResponse(
           JSON.stringify({
             client_id: "abcd",
             client_secret: "1234",
+            redirect_uris: ["https://example.com"],
           })
         ) as unknown) as Response
         /* eslint-enable @typescript-eslint/camelcase */
       );
+      const savedFetch = global.fetch;
```

Comment:
This looks a tad scary (i.e. if this tests fails after doing this). Should this be a `try { } finally { /* Restore global fetch...*/ }` ...?

Reply:
Actually, I realized that we'd never want `global.fetch` to be called in our tests before being mocked first, so in 1bd4e8d120e1fa5d1b67904fcc11a92648f39c2d I cleaned up all the saving and restoring of `global.fetch`. It should only ever be replaced by the mock which is relevant in the current test.